### PR TITLE
feat(docs): support AND security schemes

### PR DESCRIPTION
**Pull request by dotslashderek**
Part of https://github.com/stoplightio/elements/issues/2394

Just the AND support in Docs for HttpService and HttpOperation, plus test updates.

Also added some data-test attributes as @cjlanzo suggested to make e2e testing easier.

### CODE REVIEW

File: packages/elements-core/src/components/Docs/HttpOperation/Request.tsx
Comment (P0lip) [inline packages/elements-core/src/components/Docs/HttpOperation/Request.tsx:137]: That will break Proposals. We'd need to pass an id of http_security_group node here.

Comment (dotslashderek) [reply packages/elements-core/src/components/Docs/HttpOperation/Request.tsx]: Moved it over to match individual schemes again instead of the security group - I *think* this will retain current functionality - take a look :)

File: packages/elements-core/src/components/Docs/HttpOperation/Request.tsx
Comment (P0lip) [inline packages/elements-core/src/components/Docs/HttpOperation/Request.tsx:132]:
```suggestion
```
not relevant anymore :D

**Review by P0lip — COMMENTED** [review]
The code and the implementation for HttpOperation looks good to me :+1:

What about HttpService? It's currently unchanged in comparison to main

File: packages/elements-core/src/components/Docs/HttpOperation/Request.tsx
Comment (dotslashderek) [reply packages/elements-core/src/components/Docs/HttpOperation/Request.tsx:132]: 🤣

**Comment by dotslashderek** [comment]
@P0lip  - let me get that comment cleaned up :) 

And also 🤔 ... I'm going to add the optional in to this PR for HttpOperation.  That'll complete the work for endpoints.

Then I'll ask you to do another review and plan to close this PR at that point - and take all the HttpService work in a separate PR.

Reason being that this one + Optional can be done and merged directly into Elements and released now as is since they don't require any changes to http-spec and won't need to wait on the official release of stoplight/types.

Right now, since HttpService is showing the securitySchemes array instead of security, not having the AND support doesn't really matter that much?  Or, at least, it won't be apparent that Operation supports it and Service doesn't since Service as currently configured doesn't display the security property at all.

**Comment by P0lip** [comment]
> And also 🤔 ... I'm going to add the optional in to this PR for HttpOperation. That'll complete the work for endpoints.

That works, but a separate PR is fine too. :+1:
Like this already adds value, so we can ship that and add optional a bit later.

> Right now, since HttpService is showing the securitySchemes array instead of security, not having the AND support doesn't really matter that much? Or, at least, it won't be apparent that Operation supports it and Service doesn't since Service as currently configured doesn't display the security property at all.

Yeah, but we intended to switch that view to `security` as the existing behavior is just not what users expected, but I'd leave the final call to @mnaumanali94 on that as it's a solely product decision :D

**Review by P0lip — APPROVED** [review]

**Pull request merged**
